lan867x/src/esp_eth_phy_lan867x.c

```diff
@@ -0,0 +1,279 @@
+err:
+    return ret;
+}
+
+static esp_err_t lan867x_get_link(esp_eth_phy_t *phy)
+{
+    esp_err_t ret = ESP_OK;
+    phy_lan867x_t *lan867x = __containerof(esp_eth_phy_into_phy_802_3(phy), phy_lan867x_t, phy_802_3);
+
+    /* Update information about link, speed, duplex */
+    ESP_GOTO_ON_ERROR(lan867x_update_link_duplex_speed(lan867x), err, TAG, "update link duplex speed failed");
+    return ESP_OK;
+err:
+    return ret;
+}
+
+static esp_err_t lan867x_init(esp_eth_phy_t *phy)
+{
+    esp_err_t ret = ESP_OK;
+    phy_802_3_t *phy_802_3 = esp_eth_phy_into_phy_802_3(phy);
+
+    /* Basic PHY init */
+    ESP_GOTO_ON_ERROR(esp_eth_phy_802_3_basic_phy_init(phy_802_3), err, TAG, "failed to init PHY");
+    /* Check PHY ID */
+    uint32_t oui;
+    uint8_t model;
+    // to do: figure out what is wrong with this OUI layout and test it
+    ESP_GOTO_ON_ERROR(esp_eth_phy_lan867x_read_oui(phy_802_3, &oui), err, TAG, "read OUI failed");
+    ESP_GOTO_ON_ERROR(esp_eth_phy_802_3_read_manufac_info(phy_802_3, &model, NULL), err, TAG, "read manufacturer's info failed");
+    ESP_GOTO_ON_FALSE(oui == LAN867X_OUI, ESP_FAIL, err, TAG, "wrong chip OUI %lx (expected %x)", oui, LAN867X_OUI);
+
+    bool supported_model = false;
+    for (unsigned int i = 0; i < sizeof(supported_models); i++) {
+        if (model == supported_models[i]) {
+            supported_model = true;
+            break;
+        }
+    }
+    ESP_GOTO_ON_FALSE(supported_model, ESP_FAIL, err, TAG, "unsupported chip model %x", model);
+    return ESP_OK;
+err:
+    return ret;
+}
+
+static esp_err_t lan867x_autonego_ctrl(esp_eth_phy_t *phy, eth_phy_autoneg_cmd_t cmd, bool *autonego_en_stat)
+{
+    switch (cmd) {
+        case ESP_ETH_PHY_AUTONEGO_RESTART:
+        case ESP_ETH_PHY_AUTONEGO_EN:
+        case ESP_ETH_PHY_AUTONEGO_DIS:
+            return ESP_ERR_NOT_SUPPORTED; // no autonegotiation operations are supported
+        case ESP_ETH_PHY_AUTONEGO_G_STAT:
+            return 0; // since autonegotiation is not supported it is always disabled
+        default:
+            return ESP_ERR_INVALID_ARG;
+    }
+}
+
+static esp_err_t lan867x_advertise_pause_ability(esp_eth_phy_t *phy, uint32_t ability)
+{
+    return ESP_ERR_NOT_SUPPORTED;
+}
+
+static esp_err_t lan867x_set_speed(esp_eth_phy_t *phy, eth_speed_t speed)
+{
+    return ESP_ERR_NOT_SUPPORTED;
+}
+
+static esp_err_t lan867x_set_duplex(esp_eth_phy_t *phy, eth_duplex_t duplex)
+{
+    return ESP_ERR_NOT_SUPPORTED;
+}
+
+static esp_err_t lan867x_custom_ioctl(esp_eth_phy_t *phy, uint32_t cmd, void *data)
+{
+    esp_err_t ret;
+    phy_802_3_t *phy_802_3 = esp_eth_phy_into_phy_802_3(phy);
+    esp_eth_mediator_t *eth = phy_802_3->eth;
+    lan867x_plca_ctrl0_reg_t plca_ctrl0;
+    lan867x_plca_ctrl1_reg_t plca_ctrl1;
+    switch (cmd) {
+        case LAN867X_ETH_CMD_S_PLCA:
+            ESP_GOTO_ON_ERROR(eth->phy_reg_read(eth, phy_802_3->addr, ETH_PHY_PLCA_CTRL0_REG_ADDR, &(plca_ctrl0.val)), err, TAG, "read PLCA_CTRL0 failed");
```

Comment:
These registers should be accessed indirectly through the MMD Access Control (MMDCTRL) and MMD Access Address/Data (MMDAD) registers. Since MMD Access is standardized by IEEE 802.3, we should create common access function in ESP-IDF `esp_eth_phy_802_3.c`. It may get handy for other PHY's in the future.